src/common/reducers/snaps.js

```diff
@@ -0,0 +1,41 @@
+import * as ActionTypes from '../actions/snaps';
+import { parseGitHubRepoUrl } from '../helpers/github-url';
+
+export function snaps(state = {
+  isFetching: false,
+  success: false,
+  error: null,
+  snaps: null,
+}, action) {
+  switch(action.type) {
+    case ActionTypes.FETCH_SNAPS:
+      return {
+        ...state,
+        isFetching: true
+      };
+    case ActionTypes.FETCH_SNAPS_SUCCESS:
+      return {
+        ...state,
+        isFetching: false,
+        success: true,
+        snaps: action.payload.map((snap) => {
+          return {
+            // parse repository info to keep consistent data format
+            repo: parseGitHubRepoUrl(snap.git_repository_url),
```

Comment:
I thought it would be worth to keep repo info parsed already not to do it in UI components too much, but it's likely that we will only have to do it once, so maybe it's a good point that we don't need it here.

I'll remove it and if at any point we see we do this parsing in many places we can always add it back.